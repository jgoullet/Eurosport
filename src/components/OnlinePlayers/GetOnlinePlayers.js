import React, { Component } from 'react';
import './GetOnlinePlayers.scss';

class GetOnlinePlayers extends Component {
    constructor(props){
        super(props);
        this.state = {
            error : null,
            isLoaded : false,
            posts : [],    
        };
    }

    componentDidMount() {
        fetch("https://eurosportdigital.github.io/eurosport-web-developer-recruitment/headtohead.json")
        .then( response => response.json())
        .then(
            // Gére le résultat
            (result) => {
                this.setState({
                    isLoaded : true,
                    posts : result
                });
            },
            // Gére l'erreur 
            (error) => {
                this.setState({
                    isLoaded: true,
                    error
                })
            },
        )
    }


    render() {
        const {error, isLoaded, posts} = this.state;
        function countInArray(array, score) {
            var count = 0;
            for (var i = 0; i < array.length; i++) {
                if (array[i] === score) {
                    count++;
                }
            }
            return count;
        }
        if(error){
            return <div>Error in loading</div>
        }else if (!isLoaded) {
            return <div>Loading ...</div>
        }else{
            return(
                <div className="players">
                    {
                        posts.players.map((post, index)=> (
                            <div className="player" key={index}>
                                <div className="player__column-left"></div>
                                <div className="player__column-center">
                                    <img src={post.picture} alt={post.firstname + ' ' + post.lastname} />
                                    <p className="player__column-center-name"><img src={post.country.picture} alt={post.country.code} className="player__column-center-name-img-country"/> {post.firstname} {post.lastname}</p>
                                </div>
                                <div className="player__column-right">
                                    <ul className="player__column-right-info-list">
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Shortname:</div> <div className="player__column-right-info-list-item-value">{post.shortname}</div></li>
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Sex:</div> <div className="player__column-right-info-list-item-value">{post.sex}</div></li>
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Rank:</div> <div className="player__column-right-info-list-item-value">{post.data.rank}</div></li>
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Points:</div> <div className="player__column-right-info-list-item-value">{post.data.points}</div></li>
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Weight:</div> <div className="player__column-right-info-list-item-value">{post.data.weight/1000} Kg</div></li>
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Height:</div> <div className="player__column-right-info-list-item-value">{post.data.height} cm</div></li>
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Age:</div> <div className="player__column-right-info-list-item-value">{post.data.age}</div></li>
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Total wins :</div> <div className="player__column-right-info-list-item-value"></div>{countInArray(post.data.last, 1)}</li>
                                        <li className="player__column-right-info-list-item"><div className="player__column-right-info-list-item-title">Total losses :</div> <div className="player__column-right-info-list-item-value">{countInArray(post.data.last, 0)}</div></li>
                                    </ul>
                                </div>
                            </div>
                        ))
                    }
                </div>
            );
        }    
    }
  }

  export default GetOnlinePlayers;