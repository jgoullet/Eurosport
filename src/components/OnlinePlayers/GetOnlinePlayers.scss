.players{
   .player{
        padding: 1em 2em;
        color: #fff;
        display: flex;
        justify-content: center;
        align-items: center;
    &:nth-child(even){
        background-color: rgba(0,0,0,.6);
    }
    &:nth-child(odd){
        background-color: rgba(0,0,0,.7);
    }
    &__column-left{
        width: 20%;
    }
    &__column-center{
        text-align: center;
        line-height: 1rem;
        width: 50%;
        &-name{
            display: flex;
            justify-content: center;
            align-items: center;
            font-size: 2.25rem;
            line-height: 2.5rem;
            &-img-country{
                width: 2.4375rem;
                margin-right: 1.25rem;
            }
        }
    }
    &__column-right{
        width: 30%;
        display: flex;
        justify-content: center;
        &-info-list{
            padding: 1.875rem .625rem;
            background-color: rgba(255,255,255,0.1);
            text-align: left;
            list-style: none;
            width: 50%;
            &-item{
                    font-size: .875rem;
                    padding: .375rem 0;
                &-title{
                        width: 8.875rem;
                        display: inline-block;
                }
                &-value{
                    text-shadow: 1px 1px 0px #000;
                    display: inline-block;
                }
            }
        }
    }
   }

}