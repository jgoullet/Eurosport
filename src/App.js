import React, { Component } from 'react';
import GetOnlinePlayers from './components/OnlinePlayers/GetOnlinePlayers';
import './App.scss';

class App extends Component {
  render(){
  return (
    <div className="App">
      <h1 className="header">Top Players</h1>
      <GetOnlinePlayers />
    </div>
  );
  }
}

export default App;
