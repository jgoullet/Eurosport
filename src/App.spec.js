import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';

import App from './App';
import GetOnlinePlayers from './components/OnlinePlayers/GetOnlinePlayers';

describe('<App />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).to.contain(<GetOnlinePlayers />)
  })
  it('should match its reference snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).to.matchSnapshot();
  });
})