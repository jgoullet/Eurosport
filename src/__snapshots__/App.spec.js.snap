// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match its reference snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="header"
  >
    Top Players
  </h1>
  <GetOnlinePlayers />
</div>
`;
